### PR TITLE
Remove FXIOS-7004 [v117] Temp sync metrics

### PR DESCRIPTION
**Pull request by lougeniaC64**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7004)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/15540)

## :bulb: Description
We're finally removing the temp sync metrics and replacing them with the metrics that we've defined in application services. These changes include removing:
- the telemetry reporting call and the logic in `SyncTelemetryUtils.swift`
- the `pings.yaml` and `metrics.yaml` files
- the glean SDK generation configuration for the metrics

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and ensured the tests suite is passing
- [ ] Implemented accessibility and tested on UI related work (minimum Dynamic Text and VoiceOver)
- [ ] Updated documentation / comments for complex code and public methods if needed

### CODE REVIEW

File: Providers/RustSyncManager.swift [hunk 1]
```diff
@@ -189,8 +189,7 @@ public class RustSyncManager: NSObject, SyncManager {
         }
 
         if canSendUsageData() {
-            let gleanHelper = GleanSyncOperationHelper()
```
Comment (lmarceau) [inline Providers/RustSyncManager.swift:192]: It looks like `GleanSyncOperationHelper` class could be removed as well?

Comment (lougeniaC64) [reply Providers/RustSyncManager.swift:192]: @lmarceau That class is defined in the [SyncTelemetryUtils.swift](https://github.com/mozilla-mobile/firefox-ios/blob/main/Sync/SyncTelemetryUtils.swift#L12) file, which I'm deleting.

**Comment by lougeniaC64** [comment]
Feel free to leave comments, however I'm refraining from marking this PR ready for review until I'm sure I've done everything the data team requires (progress on that end can be tracked in [this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1827949)).

File: Providers/RustSyncManager.swift (hunk 1 above)
Comment (lmarceau) [reply Providers/RustSyncManager.swift:192]: Oh sorry, I missed that! Thank you!

Comment (lougeniaC64) [reply Providers/RustSyncManager.swift:192]: No problem! 😊

**Comment by lougeniaC64** [comment]
FYI: There's a bit of chicken or the egg problem here. In order for the  `Glean probe-scraper` job in this PR to pass, this [probe-scraper PR](https://github.com/mozilla/probe-scraper/pull/608) needs to be merged. However merging that PR before this one will mean that the probe scraper job in all other iOS PRs will fail (because they don't have the changes in _this_ PR).

That being the case, we're opting to merge this PR before the probe-scraper PR so that other iOS PRs aren't adversely impacted. However, this will mean that the probe scraper job in this PR will not pass (see [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1827949#c21) for details).

**Comment by lmarceau** [comment]
BR failed due to a non-related warning. Looking into that now

**Comment by lougeniaC64** [comment]
> BR failed due to a non-related warning. Looking into that now

Thanks, Laurie!

**Review by yoanarios — APPROVED** [review]

**Pull request merged**